src/tv2-common/inewsConversion/converters/ParseCue.ts

```diff
@@ -764,14 +773,17 @@ function parseAllOut(cue: string[]): CueDefinitionClearGrafiks {
 }
 
 export function parsePgmClean(cue: string[]): CueDefinitionPgmClean {
+	const pgmSource = cue[0].match(/^PGMCLEAN=(.+)$/i)
 	const pgmCleanCue: CueDefinitionPgmClean = {
 		type: CueType.PgmClean,
-		source: 'PGM',
-		iNewsCommand: 'PGMCLEAN'
+		iNewsCommand: 'PGMCLEAN',
+		sourceDefinition: { sourceType: SourceType.PGM }
 	}
-	const pgmSource = cue[0].match(/^PGMCLEAN=(.+)$/i)
 	if (pgmSource && pgmSource[1]) {
-		pgmCleanCue.source = pgmSource[1].toString().toUpperCase()
+		const sourceDefinition = getSourceDefinition(pgmSource[1])
+		if (sourceDefinition) {
```

Comment:
Would be nice to invert the condition in the if-statement on line 782 so we avoid this nested if-statement